plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergResourceFactory.java

```diff
@@ -0,0 +1,100 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.metastore.thrift.StaticMetastoreConfig;
+import io.trino.spi.TrinoException;
+import io.trino.spi.classloader.ThreadContextClassLoader;
+import io.trino.spi.connector.ConnectorSession;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.apache.iceberg.io.FileIO;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+
+public class IcebergResourceFactory
+{
+    private final StaticMetastoreConfig metastoreConfig;
+    private final IcebergConfig icebergConfig;
+
+    @Inject
+    public IcebergResourceFactory(
+            StaticMetastoreConfig metastoreConfig,
+            IcebergConfig icebergConfig)
+    {
+        this.metastoreConfig = metastoreConfig;
+        this.icebergConfig = icebergConfig;
+    }
+
+    public Catalog getTableCatalog(ConnectorSession session)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            Configuration configuration = getHadoopConfiguration();
+            Map<String, String> catalogProperties = getCatalogProperties(metastoreConfig, icebergConfig);
+            return CatalogUtil.loadCatalog(icebergConfig.getCatalogImpl(), icebergConfig.getCatalogName(), catalogProperties, configuration);
+        }
+    }
+
+    public SupportsNamespaces getSchemaCatalog(ConnectorSession session)
+    {
+        Catalog catalog = getTableCatalog(session);
+        if (catalog instanceof SupportsNamespaces) {
+            return (SupportsNamespaces) catalog;
+        }
+
+        throw new TrinoException(NOT_SUPPORTED, "catalog " + catalog.getClass().getName() + " does not support schema operations");
+    }
+
+    public FileIO getFileIO(ConnectorSession session)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            Configuration configuration = getHadoopConfiguration();
+            Map<String, String> catalogProperties = getCatalogProperties(metastoreConfig, icebergConfig);
+            return CatalogUtil.loadFileIO(icebergConfig.getIoImpl(), catalogProperties, configuration);
+        }
+    }
+
+    private static Configuration getHadoopConfiguration()
```

Comment:
This is for draft only, we need a better way to get the latest Hadoop configuration to initialize Iceberg resource. Currently the connector mostly rely on `hdfsEnvironment.getConfiguration(hdfsContext, path)`, but here we need a way to get configuration without a path. I am not very familiar with this because for Glue we can just pass in `null`. Currently I just borrowed some code from Hive's `ConfigurationUtils` to make this work.